dom.bs

```diff
@@ -266,42 +268,112 @@ against a <var>set</var>, run these steps:
+the cases where the <a>HTML parser</a> would not create an element, and otherwise canonicalizes the
+given string to equal the local name of the element the HTML parser would create.</p>
+
+<p>To <dfn>validate and canonicalize the HTML attribute local name</dfn> <var>localName</var>:</p>
 
 <ol>
- <li>If <var>namespace</var> is the empty string, set it to null.
+ <li><p>If <var>localName</var>'s <a for="string">length</a> is 0, then <a>throw</a> an
+ {{InvalidCharacterError}}.</p></li>
+
+ <li><p><a>ASCII lowercase</a> <var>localName</var>.</p></li>
+
+ <li><p>If <var>localName</var> contains <a>ASCII whitespace</a>, U+002F (/), or U+003E (>), then
+ <a>throw</a> an {{InvalidCharacterError}}.</p></li>
 
- <li><a>Validate</a> <var>qualifiedName</var>.
+ <li><p>If <var>localName</var>'s contains U+003D (=) in any position except the first character,
+ then <a>throw</a> an {{InvalidCharacterError}}.</p></li>
 
- <li>Let <var>prefix</var> be null.
+ <li><p>Replace any instances of U+0000 NULL within <var>localName</var> with U+FFFD.</p></li>
 
- <li>Let <var>localName</var> be <var>qualifiedName</var>.
+ <li><p>Return <var>localName</var>.</p></li>
+</ol>
+
+<p class="note">The <a>validate and canonicalize the HTML attribute local name</a> algorithm throws
+in the cases where the <a>HTML parser</a> would not create an attribute, and otherwise canonicalizes
+the given string to equal the local name of the attribute the HTML parser would create.</p>
 
- <li>If <var>qualifiedName</var> contains a "<code>:</code>" (U+003E), then split the
- string on it and set <var>prefix</var> to the part before and <var>localName</var> to
- the part after.
+<p>To <dfn>validate the XML local name</dfn> <var>localName</var>:</p>
+
+<ol>
+ <li><p>If <var>localName</var> does not match the <code><a type>Name</a></code> production, then
+ <a>throw</a> an {{InvalidCharacterError}}.
+</ol>
 
- <li>If <var>prefix</var> is non-null and <var>namespace</var> is null, then <a>throw</a> a
+<p>To <dfn>validate the XML qualified name</dfn> <var>qualifiedName</var>:
+
+<ol>
+ <li><p>If <var>qualifiedName</var> does not match the <code><a type>Name</a></code> or
+ <code><a type>QName</a></code> production, then <a>throw</a> an {{InvalidCharacterError}}.</p></li>
+</ol>
+
+<p>To <dfn>canonicalize a namespace and qualified name</dfn>, given <var>namespace</var>,
+<var>qualifiedName</var>, and a <var>mode</var> which is either "<code>attribute</code>" or
+"<code>element</code>":
+
+<ol>
+ <li><p>If <var>namespace</var> is the empty string, set it to null.
+
+ <li><p>Let <var>prefix</var> be null.
+
+ <li><p>Let <var>localName</var> be <var>qualifiedName</var>.
+
+ <li><p>If <var>qualifiedName</var> contains a "<code>:</code>" (U+003E), then split
+ <var>qualifiedName</var> on that first occurrence of "<code>:</code>" and set <var>prefix</var> to
+ the part before and <var>localName</var> to the part after.
```

Comment:
@domenic `before` and `after` seem a bit vague. Perhaps suffix with ... _`prefix` to the part before and `localName` to the part after the `:` respectively.

Just a thought. Sounded incomplete when reading out loud.